tools/euclideandistance_to_pickup.py

```diff
@@ -0,0 +1,62 @@
+from math import sqrt
+from entities import Container, Shipment
+from config import Config
+import numpy as np
+
+# Calculates the distance between a container location and the shipment pickup location
+# When traveling between regions, transport is only available between hubs
+# A container travels to the hub in his current region, then from that hub to the hub situated in the shipment's region,
+# and finally, from that hub to the shipment's location
+
+def euclideandistance_to_pickup(object1 : Container, object2 : Shipment):
```

Comment:
Same here. The Euclidean distance may simply be computed from say a sequence of elements (a list), through a simpler Euclidean distance method.

E.g.

```python
def euclidean_distance(x : Entity, y : Entity):
    pass


def euclidean_distance(route : List[Entity]):
    pass
```

Reply:
Both these can be used whenever needed: either for the entire route, or simpler for each subroute. This means your only need to write these things once.